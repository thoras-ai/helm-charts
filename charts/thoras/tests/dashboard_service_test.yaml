suite: Dashboard
templates:
  - dashboard/service.yaml
tests:
  - it: Default matches snapshot
    asserts:
      - matchSnapshot: {}
  - it: Works with LoadBalancer
    set:
      thorasDashboard:
        service:
          type: LoadBalancer
          LoadBalancerIP: TestIP
          loadBalancerSourceRanges:
            - Range1
            - Range2
    asserts:
      - equal:
          path: spec.type
          value: LoadBalancer
      - equal:
          path: spec.loadBalancerSourceRanges
          value:
            - Range1
            - Range2
  - it: Works with NodePort
    set:
      thorasDashboard:
        service:
          type: NodePort
    asserts:
      - equal:
          path: spec.type
          value: NodePort
  - it: Allows externalIPs
    set:
      thorasDashboard:
        service:
          externalIPs:
            - ExternalIP1
            - ExternalIP2
    asserts:
      - equal:
          path: spec.externalIPs
          value:
            - ExternalIP1
            - ExternalIP2
  - it: Allows sets ports correctly
    set:
      thorasDashboard:
        port: 5000
        nginxContainerPort: 4000
    asserts:
      - equal:
          path: spec.ports[0].port
          value: 5000
      - equal:
          path: spec.ports[0].targetPort
          value: 4000
  - it: uses correct label selector
    asserts:
      - equal:
          path: spec.selector
          value:
            app: thoras-dashboard