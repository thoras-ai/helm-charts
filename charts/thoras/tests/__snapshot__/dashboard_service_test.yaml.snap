Default matches snapshot:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      name: thoras-dashboard
      namespace: NAMESPACE
    spec:
      ports:
        - port: 80
          protocol: TCP
          targetPort: 80
      selector:
        app: thoras-dashboard
      type: ClusterIP
